packages/launch/src/focused-launch/success/index.tsx

```diff
@@ -56,9 +59,13 @@ const Success: React.FunctionComponent = () => {
 	const [ displayedSiteUrl, setDisplayedSiteUrl ] = React.useState( '' );
 	const [ hasCopied, setHasCopied ] = React.useState( false );
 
-	// if the user has an ecommerce plan or they're using focused launch from wp-admin
-	// they will be automatically redirected to /checkout, in which case the CTAs are not needed
-	const willUserBeRedirectedAutomatically = useWillRedirectAfterSuccess();
+	// Show CTA buttons needed to dismiss the success view only if the user is not going to be redirected to /checkout after launch.
+	// When we display the CTA buttons?
+	// 1. All the time for the free flow (no selected custom domain or paid plan).
+	// 2. If the selected plan isn't Ecommerce and we are in the iframed editor (conditions when we display Checkout Modal).
+	const isEcommerce = useHasEcommercePlan();
+	const isFreeFlow = ! selectedDomain && ! isSelectedPlanPaid;
+	const shouldShowCTAs = isFreeFlow || ( ! isEcommerce && isInIframe );
```

Comment:
Just noting that the same change in logic could have been achieved without refactoring the `useWillRedirectAfterSuccess` hook, which in my opinion was a nice abstraction over the implementation details of the in-editor checkout modal vs redirection.

```suggestion
	const shouldShowCTAs = isFreeFlow || ! willUserBeRedirectedAutomatically;
```

Reply:
I didn't like that hook much because:

1) It's doesn't really determine whether we'll redirect. As you can see here we need to check whether we're in free flow too.

2) If we move the free flow check inside the hook, then it will be useless [here](https://github.com/Automattic/wp-calypso/blob/38c37bbfe35b8c741c4e346637987ddc0fb3ab59/packages/launch/src/hooks/use-cart.ts#L80). Making it an abstraction that is used once.

Reply:
> It's doesn't really determine whether we'll redirect. As you can see here we need to check whether we're in free flow too.

I think that the hook represented correctly whether we'll redirect. Here we're checking for the free flow in order to show/hide CTAs (it's a different check which _also_ relies on knowing if the user will be redirected

> If we move the free flow check inside the hook, then we won't be able to use the hook here. Making it an abstraction that is used once.

We wouldn't need to move the free flow check inside the hook, because that free flow check is not needed to know if the page will be redirected — it is only needed to know if we should show the CTA buttons.

Reply:
> I think that the hook determined correctly whether we'll redirect.

I don't think this is true. The hook would return `true` for the free plan in wp-admin (`! isInIframe || ! ecommercePlan`), even though we don't redirect in this case. Right?

Reply:
Oh, now I get your previous reply too — you're correct indeed, we'd need to add the free plan check too.